go/vt/vtgate/vstream_manager.go

```diff
@@ -415,18 +415,52 @@ func (vs *vstream) alignStreams(ctx context.Context, event *binlogdatapb.VEvent,
 	}
 }
 
-func (vs *vstream) getCells() []string {
+// getCells determines the availability zones to select tablets from.
+// 2 scenarios:
+//
+// 1. No cells specified by the client via the gRPC request.
+// Tablets from the local cell of the VTGate AND the cell alias that this cell belongs to will be selected by default
+// Local cell will take precedence.
+//
+// 2. Cells are specified by the client via the gRPC request
+// These cels will take precendence over the default local cell and its alias
+// and only tablets belonging to the specified cells will be selected.
+// If the "local:" tag is passed in as an option in the list of optCells,
+// the local cell of the VTGate will take precedence over any other cell specified.
+func (vs *vstream) getCells(ctx context.Context) []string {
 	var cells []string
 	if vs.optCells != "" {
-		for _, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+		for i, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+			// if the local tag is passed in, we must give local cell priority
+			// during tablet selection. Append the VTGate's local cell to the list of cells
+			if i == 0 && cell == "local:" {
+				cells = append(cells, fmt.Sprintf("local:%s", vs.vsm.cell))
+				continue
+			}
 			cells = append(cells, strings.TrimSpace(cell))
 		}
 	}
 
+	// if no override provided in gRPC request, perform cell alias fallback
+	if len(cells) == 0 {
+		log.Info("No cells provided by client, falling back to local cell and alias...\n")
+		// append the alias this cell belongs to, otherwise appends the vtgate's cell
+		alias := topo.GetAliasByCell(ctx, vs.ts, vs.vsm.cell)
+		// an alias was actually found
+		if alias != vs.vsm.cell {
+			// send in the vtgate's cell for local cell preference
+			cells = append(cells, fmt.Sprintf("local:%s", vs.vsm.cell))
+		}
+		cells = append(cells, alias)
```

Comment:
Hmm, so this new behavior is also specific to vtgate vstreams at this point. I thought we wanted to change the default behavior of the tablet picker itself so that when no cells are specified then we use the cell alias as a fallback when we find no candidate tablets in the local cell. That's what the docs seemed to suggest and I think that we should do that.

Keep in mind that we need to document this new behavior here: https://vitess.io/docs/16.0/reference/vreplication/tablet_selection/
We should also modify it to note that it applies to vtgate vstreams as well.